elephant/statistics.py

```diff
@@ -974,23 +977,414 @@ def complexity_pdf(spiketrains, bin_size):
+            stop_bins += shift
+
+        rounding_error_indices = conv._detect_rounding_errors(start_bins,
+                                                              self.tolerance)
+
+        num_rounding_corrections = rounding_error_indices.sum()
+        if num_rounding_corrections > 0:
+            warnings.warn('Correcting {} rounding errors by shifting '
+                          'the affected spikes into the following bin. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.'.format(num_rounding_corrections))
+        start_bins[rounding_error_indices] += .5
+
+        start_bins = start_bins.astype(int)
+
+        rounding_error_indices = conv._detect_rounding_errors(stop_bins,
+                                                              self.tolerance)
+
+        num_rounding_corrections = rounding_error_indices.sum()
+        if num_rounding_corrections > 0:
+            warnings.warn('Correcting {} rounding errors by shifting '
+                          'the affected spikes into the following bin. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.'.format(num_rounding_corrections))
+        stop_bins[rounding_error_indices] += .5
+
+        stop_bins = stop_bins.astype(int)
+
+        for idx, (start, stop) in enumerate(zip(start_bins, stop_bins)):
+            time_hist[start:stop] = \
+                    self.epoch.array_annotations['complexity'][idx]
+
+        time_hist = neo.AnalogSignal(
+            signal=np.expand_dims(time_hist, axis=1),
+            sampling_period=self.bin_size, units=pq.dimensionless,
+            t_start=self.t_start)
+
+        empty_bins = (self.t_stop - self.t_start - self.epoch.durations.sum())
+        empty_bins = empty_bins.rescale(self.bin_size.units) / self.bin_size
+        if conv._detect_rounding_errors(empty_bins, tolerance=self.tolerance):
+            warnings.warn('Correcting a rounding error in the histogram '
+                          'calculation by increasing num_bins by 1. '
+                          'You can set tolerance=None to disable this '
+                          'behaviour.')
+            empty_bins += 1
+        empty_bins = int(empty_bins)
+
+        complexity_hist[0] = empty_bins
+
+        return time_hist, complexity_hist
+
+    def _epoch_no_spread(self):
+        """
+        Get an epoch object of the complexity distribution with `spread` = 0
+        """
+        left_edges = self.time_histogram.times
+        durations = self.bin_size * np.ones(self.time_histogram.shape)
+
+        if self.sampling_rate:
+            # ensure that spikes are not on the bin edges
+            bin_shift = .5 / self.sampling_rate
+            left_edges -= bin_shift
+
+            # ensure that an epoch does not start before the minimum t_start
+            min_t_start = min(st.t_start for st in self.input_spiketrains)
```

Comment:
Why do you search for the smallest t_start when all the input spike trains are checked to have the same t_start in the `__init__` constructor function?
I'll replace it with `self.t_start`. Note: I already did this in `_epoch_with_spread` function:
before
```python
min_t_start = min(st.t_start for st in self.input_spiketrains)
left_edges[0] = max(min_t_start, left_edges[0])
```
after:
```python
left_edges[0] = max(self.t_start, left_edges[0])
```

Reply:
Thanks for spotting and fixing this, seems to be a remnant of the function we started from which did not require starts and stops to be aligned.